### ISSUE TITLE
Dask joblib backend for scikit-learn models with distributed clusters overloads scheduler memory

### ISSUE THREAD

**Issue by rikturr** [dask/dask-ml#789]
I seem to be running into problems using the Dask distributed backend for joblib with scikit-learn classes. I understand this shouldn't be used for large data scenarios, but my cluster dies with relatively small sized data (50 MB).

Example taken from: https://ml.dask.org/joblib.html (I did a few `np.repeat`'s of the data to make it bigger)

It looks like this is happening because Dask makes `n` copies (one for each worker, I presume) of the data object being passed to `search.fit()`. It appears that the scheduler memory gets loaded fairly quickly which makes me think that all those copied are being pushed through the scheduler at the same time. A fix might be to send each copy directly from the client to the worker. I may be wrong about this though, and I don't know enough about how the internals of the Dask joblib backend workds. See below for how I arrived at this conclusion.

- client machine: r5.xlarge (4 cores, 32GB RAM)
- versions
```
dask                      2.30.0                     py_0    defaults
dask-cloudprovider        0.5.1                    pypi_0    pypi
dask-core                 2.30.0                     py_0    defaults
dask-glm                  0.2.0                    py37_0    defaults
dask-ml                   1.7.0                      py_0    defaults
```

This notebook has the full reproducible example using a `FargateCluster`. The issue does not happen with a `LocalCluster`: https://nbviewer.jupyter.org/gist/rikturr/66427bd13e692726044b4903a790f013

You will probably have to download the notebook and open in JupyterLab, I embedded a couple of gifs in there and github cant seem to render it

**Comment by rikturr** [comment]
Found a way to get this working where the scheduler doesn't get overloaded. It requires manually specifying the objects to be scattered like so:
```python
with joblib.parallel_backend('dask', scatter=[data, target]):
        search.fit(data, target)
```

I confirmed that scattering through the Dask client before the parallel call does _not_ work (scheduler dies like in my original example):
```python
client.scatter([data, target])
with joblib.parallel_backend('dask'):
        search.fit(data, target)
```
It appears that something with the Dask joblib backend auto-scattering causes lots of copies to go through the scheduler. An immediate "fix" is to update any dask/dask-ml documentation to explicitly scatter objects in the `parallel_backend()` call. Or we try and figure out what's going on with the auto-scattering. Ill post this to joblib also
